### PR TITLE
feat(collections): 一覧の並び順を slug / 表示名で切り替えられるようにする (#2836)

### PR DESCRIPTION
**Pull request by isamu**
## Summary

コレクション一覧に「並び順」トグル（**Slug / 表示名**）を足しました。既定は今までどおり slug 順なので、
触らないユーザーの見え方は変わりません。選択は localStorage に保存され、リロード後も保たれます。

狙いは #2836 のとおり、**slug を改名する動機を減らす**こと。表示名は dataPath と結合していないため
いつでも安全に変えられるので、「並び順を変えたい → 表示名を直す」に誘導できます。

### ソートは UI に閉じた

並び順の発生源はコードベースに 1 箇所しかありません。

```ts
// packages/core/src/collection/server/discovery.ts:288
return [...merged.values()].sort((left, right) => left.slug.localeCompare(right.slug));
```

**ここは触っていません。** `discoverCollections()` の戻り値は一覧 UI 以外にも共有されているためです。

| 消費者 | 用途 |
|---|---|
| `server/api/routes/collections.ts` | 一覧 API |
| `server/remoteHost/handlers/listCollections.ts` | モバイル（Remote Host）一覧 |
| `packages/core/src/collection-watchers/watcher.ts` | コレクション監視 |
| `packages/core/src/collection/server/ontology.ts` | Map タブ（slug 順前提） |

ソートは `CollectionsIndexView.vue` の computed の中だけで行うので、API・エージェント・監視・モバイルの
見え方は一切変わりません。

### 変更点

- `packages/plugins/collection-plugin/src/vue/collectionIndexSort.ts`（新規）— 純粋な比較子 +
  localStorage 永続化（既存 `collectionViewMode.ts` と同じ形）
- `CollectionsIndexView.vue` — トグル UI。既存のフィルタチップ行に相乗り（チップは readonly
  コレクションがある時だけ出るので、行を組み替えてトグルは常に右寄せ）。コレクションが 1 件のときは出さない
- i18n 8 ロケール（en/ja/zh/ko/es/pt-BR/fr/de）
- unit test 10 件 + e2e 4 件

## Items to Confirm / Review

1. **モバイルは対象外にしました。** Remote Host の一覧は discovery 順のままなので、desktop で表示名順に
   していても**スマホでは slug 順**に見えます。端末間で並びが割れるのを許容するか、モバイルにも同じ
   トグルを入れるか、判断をお願いします（入れるなら別 issue が妥当と考えています）。
2. **表示名を編集する UI が現状ありません。** title は skill 側の schema にあり、変更はチャット経由です。
   本 issue の狙い（slug をいじる動機を減らす）を完全に満たすには編集導線も要るかもしれません。まずは
   ソートキー切替だけ入れて様子を見る判断です。
3. **五十音順にはなりません。** issue の方針どおり辞書順のまま入れています（漢字は読み順に並ばない）。
   OS のファイル一覧と同じ挙動で、テストで固定しました。読み仮名は持たせていません。
4. **`FeedsView` は対象外**です。同じ index パターンですが今回は触っていないため、一覧間で並び基準が
   割れます（軽微と判断）。
5. **ピン留めの順序には影響しません**（確認済み）。一覧は `reconcileShortcuts()` を呼びますが、
   `src/composables/useShortcuts.ts` の `reconcile()` は既存 `shortcuts.value` の順序を保ったまま
   prune / title 更新するだけで、渡された配列の順序を採用しません。念のため渡す配列はソート前のままに
   してあります（#2519 の手動ピン順と無干渉）。
6. **リリース**: `@mulmoclaude/collection-plugin` は publish 済みパッケージです。npm 版ユーザーに届けるには
   version bump + 依存レンジ sweep + タグ付き publish が別途必要ですが、本 PR では行っていません。

## 実装上の判断

- **タイブレークは slug**。表示名が重複したときに再描画で順序が入れ替わらないようにするため。
  テストで「ソート済みを再ソートしても入れ替わらない」ことまで固定しています。
- **`Intl.Collator` に UI ロケールを明示的に渡す**。`localeCompare` の既定ロケールはブラウザ依存で、
  同じワークスペースが端末ごとに違う順に並んでしまうためです。`numeric: true` なので
  「Sprint 2 → Sprint 10」も自然順になります。
- **永続化は localStorage**（端末ごと）。サーバ設定にはしていません。既存の
  `collectionViewMode.ts`（ビューモード / テーブルのソート）と同じ扱いに揃えました。

## 検証

- `yarn format` / `yarn lint`（0 error）/ `yarn typecheck` / `yarn build` — すべて通過
- `yarn test` — 全ファイル pass、fail 0（新規 unit test 10 件を含む）
- `yarn test:e2e --grep "collections index display order"` — **4 件 pass**。実ブラウザでカードの DOM 順が
  切り替わること、リロード後も保たれること、1 件のときトグルが出ないこと、そして**ラベルが i18n から
  解決されること**（キー漏れは testid だけの断定をすり抜けるため）を確認しています。

## User Prompt

- https://github.com/receptron/mulmoclaude/issues/2836 これできる？副作用ない？
- 安全に進める

🤖 Generated with [Claude Code](https://claude.com/claude-code)

work in mulmoclaude

## Summary by Sourcery

Add a user-selectable display-order toggle to the collections index, allowing switching between slug and title-based ordering while keeping the server discovery order unchanged for non-UI consumers.

New Features:
- Introduce a Slug / Name sort toggle in the collections index to let users choose the card display order.
- Persist the selected collections index sort mode per device using localStorage.

Enhancements:
- Refactor collections index rendering to use a computed, locale-aware sorted list and hide the sort control when there is only one collection.
- Extend plugin i18n bundles with sort labels and options across supported locales.

Documentation:
- Add an internal plan document describing the collections index sorting feature, scope decisions, and release considerations.

Tests:
- Add unit tests for the collections index sort helper, covering stability, locale behavior, and invalid persisted values.
- Add Playwright E2E tests to verify UI ordering, persistence across reloads, and conditional visibility of the sort toggle.

## Summary by CodeRabbit

- **New Features**
  - Added sorting controls to collection lists, allowing sorting by slug or display name.
  - Preserves the selected sort preference across page reloads.
  - Maintains the default server-provided order when sorting by slug.
  - Applies locale-aware ordering with consistent tie-breaking.
  - Hides sorting controls when fewer than two collections are displayed.
  - Added localized sorting labels across supported languages.

- **Tests**
  - Added unit and end-to-end coverage for sorting behavior, persistence, localization, and edge cases.

### CODE REVIEW

File: packages/plugins/collection-plugin/src/vue/collectionIndexSort.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+ *  `slug` returns the fetched order untouched. Re-sorting it here would have to
+ *  reproduce discovery's `slug.localeCompare` exactly, and any drift moves cards
+ *  for users who never open the toggle — a `numeric: true` collator, for one,
+ *  puts `s2` ahead of `s10` where discovery puts `s10` first.
+ *
+ *  `title` is compared with the UI locale's collator: the browser default varies
+ *  by OS, so the same workspace would otherwise sort differently per machine.
+ *  Ties break on slug — which is unique, and compared the way discovery compares
+ *  it — so two collections sharing a title keep a stable order instead of
+ *  swapping places on every re-render. */
```
Comment (isamu) [inline packages/plugins/collection-plugin/src/vue/collectionIndexSort.ts:35]: 指摘ありがとうございます。**採用しましたが、提案とは別の直し方にしました。**

`slug` モードは比較子を差し替えるのではなく、**取得した順をそのまま返す**ようにしました（`if (key === "slug") return [...list];`）。

理由は、`<` 比較でも discovery と一致する保証が無いためです。discovery は `left.slug.localeCompare(right.slug)` を使っていて、これは ICU の照合であって符号位置順ではありません。ハイフンのような可変重み文字が入ると `<` とは結果が割れ得ます（slug は `[a-z0-9-]+` なのでハイフンは実際に出てきます）。クライアントで比較子を再現しようとすると、ズレた瞬間に今回と同じ「既定の一覧が動く」事故が再発します。素通しなら定義上ズレません。

`title` の同点タイブレークは、ご指摘どおりロケール非依存側ではなく **discovery と同じ `slug.localeCompare`** に揃えました。

回帰テストは discovery の比較子そのものをテスト内に書き、その出力と一致することを断定しています（固定値ではなくルールに対して固定）。e2e のモックも slug 順でない順序で返すようにして、既定モードがサーバ順を素通しすることを実ブラウザで押さえました。

File: test/utils/collections/test_collectionIndexSort.ts [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+  it("sorts embedded numbers naturally, not as text", () => {
+    const numbered: SortableCollection[] = [
+      { slug: "s10", title: "Sprint 10" },
+      { slug: "s2", title: "Sprint 2" },
+    ];
+    assert.deepEqual(slugsOf(sortCollectionsForIndex(numbered, "title", "en")), ["s2", "s10"]);
+  });
+
+  it("uses the locale's collator for Japanese titles — kana ahead of kanji, and kanji NOT by reading", (ctx) => {
```
Comment (isamu) [inline test/utils/collections/test_collectionIndexSort.ts:86]: ICU データ差で落ちうるという懸念は妥当なので、**ガードを入れました**。ただし提案の `Intl.Collator` モックは採っていません。

モックで並び順を固定すると、検証するのはモックの中身だけになり、「この runtime が実際にどう並べるか」は何も保証しなくなります。このテストの目的は「issue が受け入れた辞書順（漢字は読み順にならない）を仕様として固定する」ことなので、それだと意味が失われます。

代わりに、`new Intl.Collator("ja").resolvedOptions().locale !== "ja"` のとき（small-icu で ja の照合データが無く root に落ちる runtime）だけ `ctx.skip()` するようにしました。full ICU がある環境では本物の照合で断定し、無い環境では嘘の緑を出さずにスキップします。

なお現状の CI（Node 22.x / 24.x、ubuntu / macOS）はいずれも full-icu 同梱なので、実際にはスキップされずに走ります。

File: packages/plugins/collection-plugin/src/vue/collectionIndexSort.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+export function isCollectionIndexSort(value: unknown): value is CollectionIndexSort {
+  return typeof value === "string" && INDEX_SORT_KEYS.some((key) => key === value);
+}
+
+/** Order the index list by the chosen key. Titles are compared with the UI
+ *  locale's collator — the browser default varies by OS, so the same workspace
+ *  would otherwise sort differently per machine. Ties break on slug, which is
+ *  unique, so two collections sharing a title keep a stable order instead of
+ *  swapping places on every re-render. */
```
[the remark this answers is not in the document]
Comment (isamu) [reply packages/plugins/collection-plugin/src/vue/collectionIndexSort.ts:35]: 修正しました（eb5c65b）。指摘のとおりで、`numeric: true` を slug にも当てていたため `s10, s2` が `s2, s10` になり、トグルを触らないユーザーの既定表示が動いていました。

比較子を discovery に合わせるのではなく、**slug モードは取得順をそのまま返す**ようにしています。discovery の `localeCompare` をクライアントで再現しようとすると、ズレた瞬間に同じ事故が再発するためです。

回帰テストは discovery の比較子をテスト内に書き、その出力と一致することを断定します（`s10, s2, x02, x1` を含む）。e2e のモックも slug 順でない順序で返すようにして、既定モードがサーバ順を素通しすることを実ブラウザで押さえました。
